src/init.cpp

```diff
@@ -246,6 +244,10 @@ void Shutdown(NodeContext& node)
     // CValidationInterface callbacks, flush them...
     GetMainSignals().FlushBackgroundCallbacks();
 
+    // Stop all indexes only after flushing background callbacks.
```

Comment:
nit, a similar comment could be added in `BaseIndex::Stop`.

Reply:
Good point, though I think a better solution is to just call `Commit` from inside `BaseIndex::Stop` because I don't see much of a downside and it puts less of a burden on the caller.